armips/move/battle_sub_seq/345.s

```diff
@@ -0,0 +1,251 @@
+        goto DefaultOrEnd
+    goto DefaultOrEnd
+
+HandleFieldEffects:
+    ifcurrentfieldistype TERRAIN_GRASS, IsGrassField
+    ifcurrentfieldistype TERRAIN_CAVE, IsCaveField
+    goto DefaultOrEnd
+    IsGrassField:
+            if IF_EQUAL, VAR_CURRENT_MOVE, MOVE_FAIRY_WIND, GrassFieldWindBoost
+            if IF_EQUAL, VAR_CURRENT_MOVE, MOVE_SILVER_WIND, GrassFieldWindBoost
+            if IF_EQUAL, VAR_CURRENT_MOVE, MOVE_MUDDY_WATER, GrassFieldSoftenNerf
+            if IF_EQUAL, VAR_CURRENT_MOVE, MOVE_SURF, GrassFieldSoftenNerf
+            if IF_EQUAL, VAR_CURRENT_MOVE, MOVE_EARTHQUAKE, GrassFieldSoftenNerf
+            if IF_EQUAL, VAR_CURRENT_MOVE, MOVE_MAGNITUDE, GrassFieldSoftenNerf
+            if IF_EQUAL, VAR_CURRENT_MOVE, MOVE_BULLDOZE, GrassFieldSoftenNerf
+            ifmovepowergreaterthanzero GrassFieldTypeCheck
+            goto DefaultOrEnd
+        GrassFieldTypeCheck:
+            checkifcurrentadjustedmoveistype TYPE_FIRE, GrassFieldDefenderGroundedCheck
+        GrassyTerrainTypeCheck:
+            checkifcurrentadjustedmoveistype TYPE_GRASS, GrassFieldAttackerGroundedCheck
+            goto DefaultOrEnd
+        GrassFieldDefenderGroundedCheck:
+            ifgrounded BATTLER_DEFENDER, GrassFieldFireBoost
+            goto DefaultOrEnd
+        GrassFieldAttackerGroundedCheck:
+            ifgrounded BATTLER_ATTACKER, GrassFieldGrassBoost
+            goto DefaultOrEnd
+        GrassFieldFireBoost:
+            changevar VAR_OP_SET, VAR_DAMAGE_MULT, 0xD // 1.3 multiplier
+            printattackmessage
+            waitmessage
+            wait 0x1E
+            printmessage GrassFieldFireBoostMsg, TAG_NONE, "NaN", "NaN", "NaN", "NaN", "NaN", "NaN" // The grass below caught flame!
+            waitmessage
+            wait 0x1E
+            endscript
+        GrassFieldGrassBoost:
+            changevar VAR_OP_SET, VAR_DAMAGE_MULT, 0xD // 1.3 multiplier
+            printattackmessage
+            waitmessage
+            wait 0x1E
+            printmessage GrassFieldGrassBoostMsg, TAG_NONE, "NaN", "NaN", "NaN", "NaN", "NaN", "NaN" // The Grassy Terrain strengthened the attack!
+            wait 0x1E
+            endscript
+        GrassFieldWindBoost:
+            changevar VAR_OP_SET, VAR_DAMAGE_MULT, 0xD // 1.3 multiplier
+            printattackmessage
+            waitmessage
+            wait 0x1E
+            printmessage GrassFieldWindBoostMsg, TAG_NONE, "NaN", "NaN", "NaN", "NaN", "NaN", "NaN" // The wind picked up strength from the field!
+            waitmessage
+            wait 0x1E
+            endscript
+        GrassFieldSoftenNerf:
+            changevar VAR_OP_SET, VAR_DAMAGE_MULT, 0x5 // 0.5 multiplier
+            printattackmessage
+            waitmessage
+            wait 0x1E
+            printmessage GrassFieldSoftenNerfMsg, TAG_NONE, "NaN", "NaN", "NaN", "NaN", "NaN", "NaN" // The grass softened the attack...
+            waitmessage
+            wait 0x1E
+            endscript
```

Comment:
are these all vanilla interactions?

Reply:
same with all the cool interactions
while they are good ideas, i don't mean to merge them into an engine that is aiming to implement parity with later generations